src/vm/lib/metadata/agent.js

```diff
@@ -242,8 +303,9 @@ MetadataAgent.prototype.updateZone = function (zonename, callback) {
                 cb(false);
                 return;
             } else if (err) {
-                // fail open when we can't stat
-                log.error({err: err}, 'cannot fs.stat() ' + zonename);
+                // fail open (meaning: force reload) when we can't stat
+                log.error({err: err, zonename: zonename},
+                    'cannot fs.stat(), reloading');
                 cb(true);
```

Comment:
This shouldLoad() function's ultimately trying to answer a boolean question: "Should we load the VM now or not?", so the only argument the cb takes is a boolean. We could change all the calls to cb(null, [true|false]) if you think that's better. I just can't think of a case where we'd want to use the error. If stat failed here we want to reload.